### DISCUSSION TITLE
Number of operations upon website synchronisation gone up massively

### DISCUSSION THREAD

**Discussion by FLMMFL** [GetPublii/Publii#1531] [General]
hi all, 

hope someone can point me in the right direction... I have been running a site with a few hundred pages, and nothing has changed, but today when I did a website synch its gone from approx 900 operation upon website upload to 6707 operations... which massively increased the time to upload... 

1 - can you think of why the operaton increase
2 - im on version  Version: 0.44.4 (build 16557)
3 - one incident that did happen was the follow, 

i synch using sftp , and on the website url in the SFTP settings page, I accidently changed the url name to a different name, and synched which crashed my site, but having backups in place, I restored the live site within minutes... the next sync I am trying after that restore is now and its  6707 operations... which makes no sense to me... 

any thoughts?

**Comment by dziudek** [comment]
Hi,

your issue is connected with fact of change domain name - Publii is "in sync" with different address and cannot recognize your previous server instance restored from backup.

So now you have two ways:

1) If the only change was URL address and nothing changed in comparision to your site restored from backup, you can manually edit files.publii.json on your server and replace `revision` field value with value which you will find `input/config/sync-revision.json`

After this change Publii should be again "in sync" with your server. But please remember - it is only solution if your website on Publii is in the same state as website from restored backup - so if you have changed in the meantime configuration, added/deleted posts etc. then it is not a proper solution

2) If you have made some changes the best solution (especially if you have removed some posts) is to sync the website by manual deployment (ZIP icon in sync methods) - then you generate output files and upload them manually to your SFTP to replace existing files. Then revert server settings to use SFTP. Further syncs from Publii should work as usual.

FYI in Publii v.0.46 we will improve sync process and it will use list of files stored on the destination server - it will cover such edge-cases like yours, because then Publii will be harder to be in "out of sync" state.

**Reply by FLMMFL** [reply]
Morning and thank you. I think my situation is the first.. i added no posts etc, but I did add 1 small piece of code to the html, however i let it sync all 6707 operations and it completed successfully. Now when I next sync, will this just be back to normal or should I manually edit the .json file as you suggested?

**Reply by dziudek** [reply]
Hi,

After this initial full sync it should be ok for next syncs. If you will still have an issue - let me know

**Reply by FLMMFL** [reply]
ok doing I am doing a full sync again, and I noticed the sync is now 941 operations which seems closer to where it is supposed to be, but it seems very slow in the uploading to website phase. Any suggestions for speeding that up?  or is it just a measure of internet speed?  my connection is 40mb up and down on a bad day.

**Reply by dziudek** [reply]
@FLMMFL - you can check deployment process logs to make sure that there is no permission issues. If upload seems to be slow, it can suggest that the app cannot write files and retries every file few times. If changes are visible online, then probably it is something with your connection or server connection speed is slower than usual

**Reply by ghost** [reply]
I have a similar sync problem but, worse, without changing the domain name. I am using the EasyBlog template and SFTP. Every time, even if I don't make any change, Publii is replacing more than 100 files. Sometimes, about once in 10 times, Publii wants to replace all several thousand files, even if I have only made a minor change in the post text. This then results in a SFTP holdup and I have to temporarily upload the page compressed by FilleZilla and replace manually on the hosting. This is a bit annoying. I checked on two other templates on sites made from the start and there the situation is different. If I don't make any change, then Publii doesn't replace anything, and a small change in the post text causes 2-3 files to be replaced.

**Reply by jasonbrown1965** [reply]
> I have a similar sync problem but, worse, without changing the domain name. I am using the EasyBlog template and SFTP. Every time, even if I don't make any change, Publii is replacing more than 100 files. Sometimes, about once in 10 times, Publii wants to replace all several thousand files, even if I have only made a minor change in the post text. This then results in a SFTP holdup and I have to temporarily upload the page compressed by FilleZilla and replace manually on the hosting. This is a bit annoying. I checked on two other templates on sites made from the start and there the situation is different. If I don't make any change, then Publii doesn't replace anything, and a small change in the post text causes 2-3 files to be replaced.

You might want to do a separate discussion post if you want help with that.
